src/Spice86.Core/Emulator/CPU/ModRM.cs

```diff
@@ -160,4 +170,59 @@ public class ModRM {
             _ => throw new InvalidModeException(_machine, _registerMemoryIndex)
         };
     }
+
+    private int ComputeOffset32(int mode, int rm) {
+        uint result = rm switch {
+            0 => _state.EAX,
+            1 => _state.ECX,
+            2 => _state.EDX,
+            3 => _state.EBX,
+            4 => CalculateSib(mode),
+            5 => _state.EBP,
+            6 => _state.ESI,
+            7 => _state.EDI,
+            _ => throw new ArgumentOutOfRangeException(nameof(rm), rm, "Register memory index must be between 0 and 7 inclusive")
+        };
+        return (int)result;
+    }
+
+    private uint CalculateSib(int mode) {
+        byte sib = _cpu.NextUint8();
+        int scale = 1 << (sib >> 6 & 0b11);
+        int indexRegister = sib >> 3 & 0b111;
+        int baseRegister = sib & 0b111;
+        int @base = ComputeSibBase(baseRegister, mode);
+        int index = ComputeSibIndex(indexRegister);
+        return (uint)(@base + scale * index);
+    }
+
+    private int ComputeSibIndex(int indexRegister) {
+        uint result = indexRegister switch {
+            0 => _state.EAX,
+            1 => _state.ECX,
+            2 => _state.EDX,
+            3 => _state.EBX,
+            4 => 0,
```

Comment:
Isn't Index 4 is invalid?
http://www.c-jump.com/CIS77/CPU/x86/X77_0100_sib_byte_layout.htm

Reply:
No. It means "no index register is used". Setting it to 0 amounts to the same result as not using a register, since the formula is always base + (index * scale).